File: app/src/main/java/com/arcao/geocaching4locus/base/util/AnalyticsUtil.java
package com.arcao.geocaching4locus.base.util;

import com.crashlytics.android.answers.Answers;
import com.crashlytics.android.answers.CustomEvent;
import com.crashlytics.android.answers.LoginEvent;

public final class AnalyticsUtil {
    public static final String COORDINATES_SOURCE_LOCUS = "LOCUS";
    public static final String COORDINATES_SOURCE_GPS = "GPS";
    public static final String COORDINATES_SOURCE_MANUAL = "MANUAL";

    public static void actionLogin(boolean success, boolean premiumMember) {
        Answers.getInstance().logLogin(new LoginEvent().putSuccess(success)
                .putCustomAttribute("premium member", Boolean.toString(premiumMember)));
    }

    public static void actionDashboard(boolean calledFromLocus) {
        Answers.getInstance().logCustom(new CustomEvent("Dashboard")
                .putCustomAttribute("called from locus", Boolean.toString(calledFromLocus)));
    }

    public static void actionImport(boolean premiumMember) {
        Answers.getInstance().logCustom(new CustomEvent("Import")
                .putCustomAttribute("premium member", Boolean.toString(premiumMember)));
    }

    public static void actionImportBookmarks(int count, boolean all) {
        Answers.getInstance().logCustom(new CustomEvent("Import bookmarks")
                .putCustomAttribute("count", count).putCustomAttribute("all", Boolean.toString(all)));
    }

    public static void actionImportGC(boolean premiumMember) {
        Answers.getInstance().logCustom(new CustomEvent("Import GC")
                .putCustomAttribute("premium member", Boolean.toString(premiumMember)));
    }

    public static void actionSearchNearest(String coordinatesSource, boolean useFilter, int count, boolean premiumMember) {
        Answers.getInstance().logCustom(new CustomEvent("Search nearest")
                .putCustomAttribute("coordinates source", coordinatesSource == null ? COORDINATES_SOURCE_MANUAL : coordinatesSource)
                .putCustomAttribute("use filter", Boolean.toString(useFilter))
                .putCustomAttribute("count", count)
                .putCustomAttribute("premium member", Boolean.toString(premiumMember)));
    }

    public static void actionUpdate(boolean oldPoint, boolean updateLogs, boolean premiumMember) {
        Answers.getInstance().logCustom(new CustomEvent("Update")
                .putCustomAttribute("old point", Boolean.toString(oldPoint))
                .putCustomAttribute("update logs", Boolean.toString(updateLogs))
                .putCustomAttribute("premium member", Boolean.toString(premiumMember)));
    }

    public static void actionUpdateMore(int count, boolean premiumMember) {
        Answers.getInstance().logCustom(new CustomEvent("Update More")
                .putCustomAttribute("count", count)
                .putCustomAttribute("premium member", Boolean.toString(premiumMember)));
    }
}

File: app/src/main/java/com/arcao/geocaching4locus/base/util/AnalyticsUtil.kt
package com.arcao.geocaching4locus.base.util

import android.content.Context

import com.crashlytics.android.answers.Answers
import com.crashlytics.android.answers.CustomEvent
import com.crashlytics.android.answers.LoginEvent
import com.google.firebase.analytics.FirebaseAnalytics

object AnalyticsUtil {
    const val COORDINATES_SOURCE_LOCUS = "LOCUS"
    const val COORDINATES_SOURCE_GPS = "GPS"
    const val COORDINATES_SOURCE_MANUAL = "MANUAL"

    @JvmStatic
    fun actionLogin(success: Boolean, premiumMember: Boolean) {
        Answers.getInstance().logLogin(
            LoginEvent().putSuccess(success)
                .putCustomAttribute("premium member", java.lang.Boolean.toString(premiumMember))
        )
    }

    @JvmStatic
    fun actionDashboard(calledFromLocus: Boolean) {
        Answers.getInstance().logCustom(
            CustomEvent("Dashboard")
                .putCustomAttribute("called from locus", java.lang.Boolean.toString(calledFromLocus))
        )
    }

    @JvmStatic
    fun actionImport(premiumMember: Boolean) {
        Answers.getInstance().logCustom(
            CustomEvent("Import")
                .putCustomAttribute("premium member", java.lang.Boolean.toString(premiumMember))
        )
    }

    @JvmStatic
    fun actionImportBookmarks(count: Int, all: Boolean) {
        Answers.getInstance().logCustom(
            CustomEvent("Import bookmarks")
                .putCustomAttribute("count", count).putCustomAttribute("all", java.lang.Boolean.toString(all))
        )
    }

    @JvmStatic
    fun actionImportGC(premiumMember: Boolean) {
        Answers.getInstance().logCustom(
            CustomEvent("Import GC")
                .putCustomAttribute("premium member", java.lang.Boolean.toString(premiumMember))
        )
    }

    @JvmStatic
    fun actionSearchNearest(coordinatesSource: String?, useFilter: Boolean, count: Int, premiumMember: Boolean) {
        Answers.getInstance().logCustom(
            CustomEvent("Search nearest")
                .putCustomAttribute(
                    "coordinates source", coordinatesSource
                        ?: COORDINATES_SOURCE_MANUAL
                )
                .putCustomAttribute("use filter", java.lang.Boolean.toString(useFilter))
                .putCustomAttribute("count", count)
                .putCustomAttribute("premium member", java.lang.Boolean.toString(premiumMember))
        )
    }

    @JvmStatic
    fun actionUpdate(oldPoint: Boolean, updateLogs: Boolean, premiumMember: Boolean) {
        Answers.getInstance().logCustom(
            CustomEvent("Update")
                .putCustomAttribute("old point", java.lang.Boolean.toString(oldPoint))
                .putCustomAttribute("update logs", java.lang.Boolean.toString(updateLogs))
                .putCustomAttribute("premium member", java.lang.Boolean.toString(premiumMember))
        )
    }

    @JvmStatic
    fun actionUpdateMore(count: Int, premiumMember: Boolean) {
        Answers.getInstance().logCustom(
            CustomEvent("Update More")
                .putCustomAttribute("count", count)
                .putCustomAttribute("premium member", java.lang.Boolean.toString(premiumMember))
        )
    }

    @JvmStatic
    fun setPremiumUser(app: Context, premium: Boolean) {
        FirebaseAnalytics.getInstance(app).setUserProperty("premium", java.lang.Boolean.toString(premium))
    }
}
